### ISSUE TITLE
Tensorboard logging support for multi-class/multi-contrast images

### ISSUE THREAD

**Issue by leandervaneekelen** [ivadomed/ivadomed#600]
During the training of HeMIS models, I noticed that the Tensorboard logging is lacking in various facets: 

- Tensorboard images only show one class (even if the model predicts >1 class)
- Tensorboard images only show one modality (even if the model uses >1 contrast)
- 'accuracy_score' metric (orange line in plot below) seems to be stuck at ~0.99, even though Dice score (blue line) is far lower 
![image](https://user-images.githubusercontent.com/47320151/103694586-7cf6b400-4f9b-11eb-96e6-f9a78a06878c.png)

The accuracy score metric bug may have something to do with evaluation code and may be unrelated to the Tensorboard logging, but I haven't had enough time to investigate that. 

Relevant issue: #125 (other Tensorboard feature request)

**Comment by charleygros** [comment]
>     * Tensorboard images only show one class (even if the model predicts >1 class)
> 
>     * Tensorboard images only show one modality (even if the model uses >1 contrast)
> 

Both these features would be really valuable indeed!

>     * 'accuracy_score' metric (orange line in plot  below) seems to be stuck at ~0.99, even though Dice score (blue line) is far lower

Is it because the accuracy score is computed on all voxels, including background ? ie all voxel = 0 --> accuracy = 0.99. Yep this metric is not very relevant for our applications with very unbalanced classes.

**Comment by leandervaneekelen** [comment]
>Both these features would be really valuable indeed!

I'll try to integrate both features into a PR soon. 

> Is it because the accuracy score is computed on all voxels, including background?

I didn't think of that, that makes sense! Would it make sense to have an accuracy metric that is only calculated in the areas that are actually annotated? (Maybe this is too conceptually similar to dice score)

**Comment by charleygros** [comment]
> I didn't think of that, that makes sense! Would it make sense to have an accuracy metric that is only calculated in the areas that are actually annotated? (Maybe this is too conceptually similar to dice score)

I would agree: quite similar to dice score, not really needed in my perspective, but happy to discuss.

**Comment by hermancollin** [comment]
I was opening an issue for pretty much the same reasons and noticed this one.

Now I would like to bring back attention to this feature request (I could take care of this) as this would be very useful for microscopy features. Currently, training a multi-class model in ivadomed is quite impractical.

**Comment by jcohenadad** [comment]
> (I could take care of this)

🎉   please do 😊  🙏
